PennMobile/New Group/PAC Code/PacCodeViewController.swift

```diff
@@ -86,28 +86,36 @@ class PacCodeViewController : UIViewController, KeychainAccessible, ShowsAlert,
 }
 
 // MARK: - Local Authentication
-extension PacCodeViewController : LocalAuthentication {
+extension PacCodeViewController : KeychainAccessible, LocallyAuthenticatable {
         
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         
-        handleAuthentication(cancelText: "Go Back", reasonText: "Authenticate to see your PAC Code")
+        requestAuthentication(cancelText: "Go Back", reasonText: "Authenticate to see your PAC Code")
     }
     
     override func viewDidDisappear(_ animated: Bool) {
         super.viewDidDisappear(animated)
         pacCode = nil
-        refreshPacCode()
+        updatePACCode()
     }
     
     func handleAuthenticationSuccess() {
         if let pacCode = getPacCode() {
             self.pacCode = pacCode
         } else {
-            self.showAlert(withMsg: "Please login to use this feature", title: "Login Error", completion: { self.navigationController?.popViewController(animated: true)} )
+            if Account.isLoggedIn {
+                self.showActivity()
+                
+                // Handle the case in which the user is logged in but hasn't yet fetched their PAC Codes
+                // Acts as though the user pressed the refresh button
+                PacCodeNetworkManager.instance.getPacCode { result in self.handleNetworkPacCodeRefreshCompletion(result) }
+            } else {
+                self.showAlert(withMsg: "Please login to use this feature", title: "Login Error", completion: { self.navigationController?.popViewController(animated: true)} )
+            }
         }
         
-        refreshPacCode()
+        updatePACCode()
```

Comment:
Let's only call this function if the pacCode is not nil. No reason to call it otherwise.